wagtail/snippets/views/snippets.py

```diff
@@ -1151,9 +1177,26 @@ def get_urlpatterns(self):
 
         return urlpatterns + legacy_redirects
 
+    def register_admin_url_finder(self):
+        register_admin_url_finder(self.model, self.url_finder_class)
+
+    def register_chooser_widget(self):
+        # Set up admin model forms to use AdminSnippetChooser for any ForeignKey to this model
+        register_form_field_override(
+            ForeignKey,
+            to=self.model,
+            override={"widget": AdminSnippetChooser(model=self.model)},
+        )
```

Comment:
Alternatively, we can override `SnippetChooserViewSet.widget_class` and unset `register_widget = False` there

Reply:
Actually yeah, I think that seems better. I've changed it to use that approach.